### ISSUE TITLE
Rules: `semi-spacing` and `block-spacing` conflict on one-liner functions

### ISSUE THREAD

**Issue by Cellule** [eslint/eslint#3721]
With the following rules, there is no way to have a function in one line without a conflict between the 2.
```js
// eslint v1.3.1
/*eslint "semi-spacing":2*/
/*eslint "block-spacing":[2, "never"]*/
var fnArr = [
  function foo(){return 2;}, //4:26  error  Missing whitespace after semicolon  semi-spacing
  function foo2(){return 2; }, //5:29  error  Unexpected space(s) before "}"      block-spacing
  function foo3(){
    return 2; // No errors, but this is not what I am trying to do
  }
]
```

Is there a way to have both rule and a function declaration in one line ?

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by nzakas** [comment]
Hmm, we need to figure out how to clean that up. I'm not sure how at the moment.

**Comment by Cellule** [comment]
I just saw a similar issue with `space-in-parens`
where 
`for(let i = 0; i < results.length;) {` is invalid because of `semi-spacing` and
`for(let i = 0; i < results.length; ) {` is invalid because of `space-in-parens`
The only valid case is 
```js
for(let i = 0; i < results.length; 
) {
```

**Comment by mysticatea** [comment]
So, should `semi-spacing` be skipped if the next token is a closing paren (`)`, `}`, `]`)?

**Comment by nzakas** [comment]
That might be the solution. We'd have to update the docs to reflect that as well.

**Comment by mysticatea** [comment]
I will work on this.
